python/lsst/meas/algorithms/stamps.py

```diff
@@ -334,12 +344,10 @@ def readFitsWithOptions(cls, filename, options):
         """
         # To avoid problems since this is no longer an abstract method
         if cls is not StampsBase:
-            raise NotImplementedError(
-                f"Please implement specific FITS reader for class {cls}"
-            )
+            raise NotImplementedError(f"Please implement specific FITS reader for class {cls}")
```

Comment:
Either here, or in a future ticket (comment it here with a TODO in that case), this method should be refactored. The logic looks very convoluted and I don't follow what's going on here. If `readFitsWithOptions` is called on `StampBase`, then something happens but if called on a child class (Stamps), then an error is raised? And `cls` is overridden below, but why not call the method directly on `stamp_type`?

Reply:
I think it's related to the trampoline to allow the butler formatter to use a generic type but still end up giving you the correct type and making sure that you don't by mistake end up trying to use the abstract base class for reading.

Reply:
I've added this in-line comment, which I hope captures the gist of the discussion above:
```
        # TO-DO: Consider refactoring this method. This class check was added
        # to allow the butler formatter to use a generic type but still end up
        # giving the correct type back, ensuring that the abstract base class
        # is not used by mistake. Perhaps this logic can be optimised.
```